False positive with go test -trace=test.out

As stated in the title, when you invoke the test command with a trace option, the test always fails with a message concerning trace:

> --- FAIL: TestAnd (0.45s)
    leaks.go:78: found unexpected goroutines:
        [Goroutine 6 in state trace reader (blocked), with runtime.goparkunlock on top of the stack:
        goroutine 6 [trace reader (blocked)]:
        runtime.goparkunlock(...)
                c:/go/src/runtime/proc.go:310

Even when the test is empty like this one :

	package main
	
	import (
		"testing"
	
		"go.uber.org/goleak"
	)
	
	func TestAnd(t *testing.T) {
		defer goleak.VerifyNone(t)
	
	}

--- a/Makefile
+++ b/Makefile
@@ -17,6 +17,7 @@ install:
 .PHONY: test
 test:
 	go test -v -race ./...
+	go test -v -trace=/dev/null .
 
 .PHONY: cover
 cover:

--- a/options.go
+++ b/options.go
@@ -78,6 +78,7 @@ func buildOpts(options ...Option) *opts {
 		isTestStack,
 		isSyscallStack,
 		isStdLibStack,
+		isTraceStack,
 	)
 	for _, option := range options {
 		option.apply(opts)
@@ -140,4 +141,12 @@ func isStdLibStack(s stack.Stack) bool {
 
 	// Using signal.Notify will start a runtime goroutine.
 	return strings.Contains(s.Full(), "runtime.ensureSigM")
+}
+
+func isTraceStack(s stack.Stack) bool {
+	if f := s.FirstFunction(); f != "runtime.goparkunlock" {
+		return false
+	}
+
+	return strings.Contains(s.Full(), "runtime.ReadTrace")
 }

